meson.build

```
@@ -1394,36 +1394,38 @@ if features['rpi-mmal']
                      'video/out/vo_rpi.c')
 endif
 
-vaapi = dependency('libva', version: '>= 1.1.0', required: get_option('vaapi'))
-features += {'vaapi': vaapi.found() and features['libdl'] and (features['x11'] or
-             features['wayland'] or egl_drm.allowed())}
+libva = dependency('libva', version: '>= 1.1.0', required: get_option('vaapi'))
 
-if features['vaapi']
-    dependencies += vaapi
-    sources += files('video/filter/vf_vavpp.c',
-                     'video/vaapi.c')
-endif
-
-vaapi_drm = dependency('libva-drm', version: '>= 1.1.0', required: get_option('vaapi-drm').require(features['vaapi']))
-features += {'vaapi-drm': features['vaapi'] and vaapi_drm.found()}
+vaapi_drm = dependency('libva-drm', version: '>= 1.1.0', required: get_option('vaapi-drm').require(libva.found() and features['drm']))
```

Sorry just one last style thing. These lines are all pretty long could you break them into two like this?
```suggestion
vaapi_drm = dependency('libva-drm', version: '>= 1.1.0',
                                          required: get_option('vaapi-drm').require(libva.found() and features['drm']))
```